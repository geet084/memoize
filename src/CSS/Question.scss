@mixin colorSet($color, $bgc) {
  color: $color;
  background-color: $bgc;
}

@mixin inputFormat($width, $height, $margin) {
  width: $width;
  height: $height;
  margin: $margin;
}

.count {
  color: #ffeb3b;
}

.definition {
  width: 75%;
  height: 110px;
  margin: 5px;
  padding: 10px;
  color:#013309;
  background-color: #add9ec;
  opacity: .65;
  border-radius: 25px;
  font-size: 1.75rem;
}

.num-guess {
  @include colorSet(#fff, #016FB9);
  width: 40%;
  margin: 20px 5px 5px 5px;
  padding: 10px;
  border-radius: 15px;
}

.show-previous {
  @include colorSet(#fff, #016FB9);
  width: 40%;
  margin: 5px;
  padding: 10px;
  border-radius: 15px;
}

.question {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
}

.no {
  color: maroon;
}

.yes {
  color: lightgreen;
}

.undefined {
  display: none;
}

.user-btn {
  @include inputFormat(290px, 35px, 10px);
  border-radius: 10px;
  font-size: 1.25rem;
  color: #fff;
  background-color: #016FB9;
  &:hover {
    @include colorSet(#013309, #add9ec);
  }
}

.user-input {
  @include inputFormat(290px, 35px, 10px);
  padding: 5px 2px 5px 5px;
  font-size: 1.25rem;
  
  background-color: #016FB9;
  &::placeholder {
    color:#fff;
  }
  &:hover {
    @include colorSet(#013309, #add9ec);
    &::placeholder {
      color: #013309;
    }
  }
}

