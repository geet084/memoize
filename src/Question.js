import React, { Component } from 'react';
import './CSS/Main.scss';

export default class Question extends Component {

  render() {
    let { count, definition, previous, isCorrect, checkAnswer, showPrevResult, showBtnText } = this.props;
    return (
      <div className="question">
        <p className="num-guess">
          Number of guesses so far:
            <span className="count">  {count}</span>
        </p>
        <p className="definition">{definition}</p>
        <p className={previous}>
          Previous result for this question was:
            <span className={isCorrect}>
            {showPrevResult}
          </span>
        </p>
        <input type="text" className="user-input"
          placeholder="ENTER YOUR ANSWER HERE"
          onFocus={(e) => e.target.placeholder = ""}
          onBlur={(e) => e.target.placeholder = "ENTER YOUR ANSWER HERE"} />
        <button onClick={checkAnswer}
          className="user-btn">{showBtnText}
        </button>
      </div>
    )
  }
}