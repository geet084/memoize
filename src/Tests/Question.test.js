import React from 'react';
import Question from '../Question.js';
import { shallow } from 'enzyme';

const mockCount = 0;
const mockDefinition = undefined;
const mockPrevious = 'show-previous';
const mockIsCorrect = 'yes';
const mockShowPrevResult = jest.fn();
const mockShowBtnText = jest.fn();
const mockCheckAnswer = jest.fn();

describe('Question', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Question
        count={mockCount}
        definition={mockDefinition}
        previous={mockPrevious}
        isCorrect={mockIsCorrect}
        showPrevResult={mockShowPrevResult}
        showBtnText={mockShowBtnText}
        checkAnswer={mockCheckAnswer} />
    )
  })

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should register when clicked for checking the answer', () => {

    wrapper.find('.user-btn').simulate('click');
    expect(mockCheckAnswer).toBeCalled();
  })
});

