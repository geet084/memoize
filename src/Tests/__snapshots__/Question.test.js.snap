// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question should match the snapshot with all data passed in correctly 1`] = `
<div
  className="question"
>
  <p
    className="num-guess"
  >
    Number of guesses so far:
    <span
      className="count"
    >
        
      0
    </span>
  </p>
  <p
    className="definition"
  />
  <p
    className="show-previous"
  >
    Previous result for this question was:
    <span
      className="yes"
    >
      <Component />
    </span>
  </p>
  <input
    className="user-input"
    onBlur={[Function]}
    onFocus={[Function]}
    placeholder="ENTER YOUR ANSWER HERE"
    type="text"
  />
  <button
    className="user-btn"
    onClick={[MockFunction]}
  >
    <Component />
  </button>
</div>
`;
